bigbluebutton-html5/imports/ui/components/video-dock/component.jsx

```diff
@@ -98,22 +121,32 @@ export default class VideoDock extends Component {
   componentWillMount () {
     this.ws.addEventListener('open', this.onWsOpen);
     this.ws.addEventListener('close', this.onWsClose);
+
+    window.addEventListener('online', this.ws.open.bind(this.ws));
+    window.addEventListener('offline', this.ws.close.bind(this.ws));
   }
 
   componentWillUnmount () {
     document.removeEventListener('joinVideo', this.shareWebcam);
-    document.removeEventListener('exitVideo', this.shareWebcam);
+    document.removeEventListener('exitVideo', this.unshareWebcam);
+    document.removeEventListener('installChromeExtension', this.installChromeExtension);
     window.removeEventListener('resize', this.adjustVideos);
 
     this.ws.removeEventListener('message', this.onWsMessage);
     this.ws.removeEventListener('open', this.onWsOpen);
     this.ws.removeEventListener('close', this.onWsClose);
     // Close websocket connection to prevent multiple reconnects from happening
+
+    window.removeEventListener('online', this.ws.open);
+    window.removeEventListener('offline', this.ws.close);
+
     this.ws.close();
   }
 
   adjustVideos () {
-    window.adjustVideos('webcamArea', true);
+    setTimeout(() => {
```

Comment:
👀

Reply:
I know it sucks, but we need to wait a tick here because in some cases the webcamArea does not yet have a different size when the 'resize' event was propagated.

Reply:
It's fine.
We have `resize` events dispatched in "componentDidMount" and in "componentWillUnmount"..
So the first ones comes **after** the chat was opened, for instance. But the event from "componentWillMount" is dispatched **before** the chat is closed.
`setTimeout` just puts the action on top of the stack, waiting for all the pending ui changes to be completed.